evcouplings/couplings/mean_field.py

```diff
@@ -549,6 +589,175 @@ def to_raw_ec_file(self, couplings_file):
                         "{0:.6f}".format(self.di_scores[i, j])
                     ])) + "\n")
 
+    def adapt_from_plmc_reading(self):
+        """
+        Adaptions that allow to read
+        a mean-field couplings model
+        from file where __read_plmc_v2
+        in CouplingsModel does the
+        heavy lifting.
+
+        This includes:
+        - Manage unused plmc-specific
+        fields as well as the pseudo count
+        field
+        - Modify pair frequencies
+        - Regularize column and pair
+        frequencies (i.e. fill the fields
+        regularized_f_i and regularized_f_ij)
+        """
+        self.__decode_unused_fields()
+
+        # set the frequency of a pair (alpha, alpha)
+        # in position i to the respective single-site
+        # frequency of alpha in position i
+        for i in range(self.L):
+            for alpha in range(self.num_symbols):
+                self.f_ij[i, i, alpha, alpha] = self.f_i[i, alpha]
+
+        # compute pseudo counted frequencies
+        # from raw frequencies
+        self.regularize_f_i()
+        self.regularize_f_ij()
+
+    def __encode_unused_fields(self):
+        """
+        Set plmc-specific parameters
+        (lambda_J, lambda_group, num_iter)
+        to an arbitrary numerical value.
+
+        Note:
+        The field lambda_h is used to store
+        the pseudo count (the negative sign
+        simply serves as marker).
+        """
+        self.lambda_J = -1
+        self.lambda_group = -1
+        self.num_iter = -1
+        self.lambda_h = -self.pseudo_count
+
+    def __decode_unused_fields(self, save_pseudo_count=True):
+        """
+        Set plmc-specific parameters
+        to None to ensure correct usage
+        of the object.
+
+        Note:
+        If save_pseudo_count is True,
+        the pseudo count (that was temporarily
+        stored in lambda_h) is stored in the
+        appropriate field.
+
+        Parameters
+        ----------
+        save_pseudo_count : bool, optional (default: True)
+            If True, the pseudo count is read from
+            the field lambda_h and stored in the
+            appropriate field.
+        """
+        self.lambda_J = None
+        self.lambda_group = None
+        self.num_iter = None
+
+        if save_pseudo_count:
```

Comment:
I would remove the if clause and always execute `self.pseudo_count = -self.lambda_h` (unless I am missing something here?)... otherwise if one calls with save_pseudo_count=False, the encode function might blow up because the variable is not defined

Reply:
I came up with that because I wanted to use the __decode_unused_fields function in the MeanFieldCouplingsModel constructor. At that time, self.lambda_h is not set.

Reply:
Ah I see - I think this is a valid solution (alternatively hasattr comes to mind)